recipes/configure.rb

```diff
@@ -0,0 +1,60 @@
+# encoding: UTF-8
+# Cookbook Name:: apache_zookeeper
+# Re
+# Recipe:: configure
+
+# Create all the directories
+#class ::Chef::Recipe
+#  include ::ZookeeperHelper
+#end
+class ::Chef::Recipe
+  include ::ZookeeperHelper
+end
+class ::Chef::Resource
+  include ::ZookeeperHelper
+end
+
+[
+  node['apache_zookeeper']['config_dir'],
+  node['apache_zookeeper']['bin_dir'],
+  node['apache_zookeeper']['data_dir'],
+  node['apache_zookeeper']['log_dir'],
+].each do |dir|
+  directory dir do
+    recursive true
+    owner node['apache_zookeeper']['user']
+  end
+end
+
+template 'log4j properties' do
+  path ::File.join(node['apache_zookeeper']['config_dir'], 'log4j.properties')
+  group node['apache_zookeeper']['group']
+  owner 'root'
+  mode  00644
+  backup false
+  source 'log4j.properties.erb'
+  #notifies :restart, 'service[zookeeper]'
```

Comment:
This only works if you run the service recipe too. Is it possible to put a
service entry in both recipes? I could give that a try
On Jul 24, 2015 6:56 AM, "Bryan Baugher" <notifications@github.com> wrote:

> In recipes/configure.rb
> <https://github.com/bbaugher/apache_zookeeper/pull/29#discussion_r35415772>
> :
>
> > +  node['apache_zookeeper']['log_dir'],
> > +].each do |dir|
> > +  directory dir do
> > +    recursive true
> > +    owner node['apache_zookeeper']['user']
> > +  end
> > +end
> > +
> > +template 'log4j properties' do
> > +  path ::File.join(node['apache_zookeeper']['config_dir'], 'log4j.properties')
> > +  group node['apache_zookeeper']['group']
> > +  owner 'root'
> > +  mode  00644
> > +  backup false
> > +  source 'log4j.properties.erb'
> > +  #notifies :restart, 'service[zookeeper]'
>
> Was this giving you problems? I would still think we need it and the one
> below
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bbaugher/apache_zookeeper/pull/29/files#r35415772>.
>